BinaryTreeNodeTest.java

```diff
@@ -0,0 +1,76 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BinaryTreeNodeTest {
+    private BinaryTreeNode<Integer> n1, n2, n3, n4, n5, n6, n7, n8;
+
+    @BeforeEach
+    void setUp() {
+        // Set up the tree
+        n1 = new BinaryTreeNode<>(1, null, null);
+        n2 = new BinaryTreeNode<>(5, null, null);
+        n3 = new BinaryTreeNode<>(3, n1, n2);
+        n4 = new BinaryTreeNode<>(14, null, null);
+        n5 = new BinaryTreeNode<>(9, n3, n4);
+        n6 = new BinaryTreeNode<>(19, null, null);
+        n7 = new BinaryTreeNode<>(18, null, n6);
+        n8 = new BinaryTreeNode<>(16, n5, n7);
+    }
+
+    @Test
+    void findAncestorsTest() {
+        // Normal case
+        assertEquals(n8.findAncestors(18), Collections.singletonList(16));
+
+        // Leaf case
+        assertEquals(n8.findAncestors(5), Arrays.asList(3, 9, 16));
+
+        // Root case
+        assertEquals(n8.findAncestors(16), Collections.emptyList());
+
+        // Not in tree case
+        assertEquals(n8.findAncestors(2555), null);
+
+        // Null case
+        assertEquals(n8.findAncestors(null), null);
+    }
+
+    @Test
+    void findCommonAncestorTest() {
```

Comment:
Like above, could you please split it to separate one-assertion methods, each annotated with `@Test`?